### PR TITLE
NEML2Action overhaul

### LINKED ISSUE #27553: Improve modular NEML2 interface
### Motivation

A use case for nodal NEML2 model execution has popped up, and the execution during the `INITIAL` execution state needs to be enabled. Investigate micro optimizations for tensor conversion and contiguous storage for fast memcopy transfers back to MOOSE. Add some more error checking.

### Design

Template `ExecuteNEML2Model` and add a nodal version. Figure out why the `INITIAL` stage is not properly executing. Check the compatibility of block restrictions of the involved user objects.

### Impact

Added capability

### PR DESCRIPTION
**Pull request by hugary1995**
- Factor out a common action for NEML2 for block-restriction
- Remove legacy objects
- Unify gatherers for variable and parameter
- Migrate all input files to use NEML2Action
- Remove legacy tests

close #27553

### CODE REVIEW

**Comment by hugary1995** [comment]
@dschwen @reverendbedford @bwspenc, I'd like to get some early feedback on the input file syntax. See input files in
```
modules/solid_mechanics/test/tests/neml2
```

Gathering MOOSE material properties as NEML2 model parameters is supported, but I haven't prepared an input file for it.

I also talked to @dschwen regarding splitting `ExecuteNEML2Model` into a `GeneralUserObject` and an `ElementUserObject` -- that's on my to-do list for this PR.

I need to work on something else for the rest of this week. I will come back to this PR next week.

**Comment by hugary1995** [comment]
@dschwen on a related note, I also added support for transferring MOOSE postprocessor value to NEML2.

File: modules/solid_mechanics/test/tests/neml2/viscoplasticity_chaboche.i [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    type = RankTwoTensorToSymmetricRankTwoTensor
+    from = 'mechanical_strain'
+    to = 'neml2_strain'
+  []
+  [stress]
+    type = ComputeLagrangianObjectiveCustomSymmetricStress
+    custom_small_stress = 'neml2_stress'
+    custom_small_jacobian = 'neml2_jacobian'
+  []
+[]
```
Comment (hugary1995) [inline modules/solid_mechanics/test/tests/neml2/viscoplasticity_chaboche.i:71]: Here's the beef.

Comment (dschwen) [reply modules/solid_mechanics/test/tests/neml2/viscoplasticity_chaboche.i:71]: Next FY I'd love to gett embedded neml2 syntax working. That would enable us to change neml2 object options on the command line and use common brace expansions.

**Review by dschwen — APPROVED** [review]
This is nice. Should have reviewed much earlier. Apologies!

**Pull request merged**